src/coreclr/System.Private.CoreLib/src/System/Buffer.CoreCLR.cs

```diff
@@ -16,7 +17,13 @@ public partial class Buffer
         private static unsafe partial void __Memmove(byte* dest, byte* src, nuint len);
 
         [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern void __BulkMoveWithWriteBarrier(ref byte destination, ref byte source, nuint byteCount);
+        private static extern void BulkMoveWithWriteBarrierInternal(ref byte destination, ref byte source, nuint byteCount);
+
+        private static void __BulkMoveWithWriteBarrier(ref byte destination, ref byte source, nuint byteCount)
+        {
+            Thread.PollGC();
+            BulkMoveWithWriteBarrierInternal(ref destination, ref source, byteCount);
```

Comment:
Unlike most FCall, BulkMoveWithWriteBarrier is longer-running FCall, so the risk of GC starvation is more real for it. We should try to do something about that.

Reply:
Is calling the managed `PollGC` not sufficient here? I could also convert this to a QCall with `SuppressGCTransition` and see the impact.

Reply:
It is sufficient, but it likely comes with significant overhead since it is not understood by the JIT as intrinsic.

Reply:
Looks like between 20 % and 30 % impact, but very small. Perhaps converting to QCall would be better?